### PR TITLE
Adding boolean conditional expressions, numeric conditional expressions to conditional expressions resolution

### PR DESCRIPTION
**Pull request by jlerbsc**
Adding boolean conditional expressions, numeric conditional expressions to conditional expressions resolution

### CODE REVIEW

File: javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/promotion/NumericConditionalExprHandler.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    ResolvedType elseExpr;
+
+    public NumericConditionalExprHandler(ResolvedType thenExpr, ResolvedType elseExpr) {
+        this.thenExpr = thenExpr;
+        this.elseExpr = elseExpr;
+    }
+
+    @Override
+    public ResolvedType resolveType() {
+        String qnameTypeThenExpr = thenExpr.isPrimitive() ? thenExpr.asPrimitive().describe()
```
Comment (matozoid) [inline javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/promotion/NumericConditionalExprHandler.java:43]: I think we have "ifPrimitive" too, which gives an Optional, and then you can use map().orElse()

File: javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/AbstractSymbolResolutionTest.java [hunk 2]
```diff
@@ -21,13 +21,24 @@
...
+import org.junit.jupiter.api.AfterEach;
+
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.utils.CodeGenerationUtils;
+
 public abstract class AbstractSymbolResolutionTest {
+    
+    @AfterEach
+    public void reset() {
```
Comment (matozoid) [inline javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/AbstractSymbolResolutionTest.java:37]: I decided to create a test parser somewhere, I was using that lately. I'm not sure what it is called. Better not use StaticJavaParser for the reason you mention.

File: javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/AnnotationsResolutionTest.java [hunk 3]
```diff
@@ -21,6 +21,16 @@
 
 package com.github.javaparser.symbolsolver.resolution;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
```
Comment (matozoid) [inline javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/AnnotationsResolutionTest.java:27]: There is a formatter settings file in the repository. (Ignore this if the formatting of this file was incorrect :-) )

Comment (jlerbsc) [reply javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/AnnotationsResolutionTest.java:27]: Done.

**Comment by matozoid** [comment]
Why not merge this? The list of PR's is enormous :-)

**Pull request closed without merging**